File: imageslider/src/main/java/technolifestyle/com/imageslider/FlipperAdapter.java
package technolifestyle.com.imageslider;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

class FlipperAdapter extends PagerAdapter {

    private Context context;
    private ArrayList<FlipperView> flipperViews = new ArrayList<>();

    FlipperAdapter(Context context) {
        this.context = context;
    }

    public void setFlipperViews(ArrayList<FlipperView> flipperViews) {
        this.flipperViews = flipperViews;
    }

    void addFlipperView(FlipperView view) {
        flipperViews.add(view);
        notifyDataSetChanged();
    }

    public void removeAllFlipperViews() {
        flipperViews.clear();
        notifyDataSetChanged();
    }

    public FlipperView getFlipperView(int position) {
        if (flipperViews.isEmpty() || position >= flipperViews.size()) {
            return null;
        }
        return flipperViews.get(position);
    }

    @Override
    public int getCount() {
        return flipperViews.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == object;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        FlipperView imageFlipperView = flipperViews.get(position);
        View v = imageFlipperView.getView();
        container.addView(v);
        return v;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((View) object);
    }
}

File: imageslider/src/main/java/technolifestyle/com/imageslider/FlipperAdapter.kt
package technolifestyle.com.imageslider

import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import java.util.*

internal class FlipperAdapter : PagerAdapter() {
    private var flipperViews = ArrayList<FlipperView>()

    fun setFlipperViews(flipperViews: ArrayList<FlipperView>) {
        this.flipperViews = flipperViews
    }

    fun addFlipperView(view: FlipperView) {
        flipperViews.add(view)
    }

    fun removeAllFlipperViews() {
        flipperViews.clear()
        notifyDataSetChanged()
    }

    fun getFlipperView(position: Int): FlipperView? {
        return if (flipperViews.isEmpty() || position >= flipperViews.size) {
            null
        } else flipperViews[position]
    }

    override fun getCount(): Int {
        return flipperViews.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val imageFlipperView = flipperViews[position]
        val view = imageFlipperView.getView()
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as View)
    }

    fun addAllFlipperViews(flipperViewList: List<FlipperView>) {
        flipperViews.addAll(flipperViewList)
        notifyDataSetChanged()
    }
}
